src/NuGet.Core/NuGet.Build.Tasks.Console/DependencyGraphSpecGenerator.cs

```diff
@@ -0,0 +1,893 @@
+        /// <returns>A <see cref="List{TargetFrameworkInformation}" /> containing the target framework information for the specified project.</returns>
+        private static List<TargetFrameworkInformation> GetTargetFrameworkInfos(IReadOnlyDictionary<NuGetFramework, IMSBuildProject> projectInnerNodes)
+        {
+            var targetFrameworkInfos = new List<TargetFrameworkInformation>(projectInnerNodes.Count);
+
+            foreach (var projectInnerNode in projectInnerNodes)
+            {
+                var msBuildProjectInstance = projectInnerNode.Value;
+
+                var targetFrameworkInformation = new TargetFrameworkInformation
+                {
+                    FrameworkName = projectInnerNode.Key,
+                    RuntimeIdentifierGraphPath = msBuildProjectInstance.GetProperty(nameof(TargetFrameworkInformation.RuntimeIdentifierGraphPath))
+                };
+
+                var packageTargetFallback = MSBuildStringUtility.Split(msBuildProjectInstance.GetProperty("PackageTargetFallback")).Select(NuGetFramework.Parse).ToList();
+
+                var assetTargetFallback = MSBuildStringUtility.Split(msBuildProjectInstance.GetProperty(nameof(TargetFrameworkInformation.AssetTargetFallback))).Select(NuGetFramework.Parse).ToList();
+
+                AssetTargetFallbackUtility.EnsureValidFallback(packageTargetFallback, assetTargetFallback, msBuildProjectInstance.FullPath);
+
+                AssetTargetFallbackUtility.ApplyFramework(targetFrameworkInformation, packageTargetFallback, assetTargetFallback);
+
+                targetFrameworkInformation.Dependencies.AddRange(GetPackageReferences(msBuildProjectInstance));
+
+                targetFrameworkInformation.DownloadDependencies.AddRange(GetPackageDownloads(msBuildProjectInstance));
+
+                targetFrameworkInformation.FrameworkReferences.AddRange(GetFrameworkReferences(msBuildProjectInstance));
+
+                targetFrameworkInfos.Add(targetFrameworkInformation);
+            }
+
+            return targetFrameworkInfos;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="DependencyGraphSpec" /> for the specified project.
+        /// </summary>
+        /// <param name="entryProjectPath">The full path to a project or Visual Studio Solution File.</param>
+        /// <param name="globalProperties">An <see cref="IDictionary{String,String}" /> containing the global properties to use when evaluation MSBuild projects.</param>
+        /// <returns>A <see cref="DependencyGraphSpec" /> for the specified project if they could be loaded, otherwise <code>null</code>.</returns>
+        private DependencyGraphSpec GetDependencyGraphSpec(string entryProjectPath, IDictionary<string, string> globalProperties)
+        {
+            try
+            {
+                // TODO: Use a localized resource from https://github.com/NuGet/NuGet.Client/pull/3111
```

Comment:
I was trying to avoid having to handle the merge conflict later.  Am I expected to open issues for code that hasn't been merged yet?  My plan was as part of the other PR to update this code path to use the new localized resource.